operators/api/v1alpha1/tenant_types.go

```diff
@@ -23,65 +23,89 @@ import (
 
+// WorkspaceUserRole is an enumeration of the different roles that can be
+// associated to a Tenant in a Workspace.
 // +kubebuilder:validation:Enum=manager;user
 type WorkspaceUserRole string
 
 const (
-	// Manager allows to interact with all VMs of a workspace
+	// Manager -> a Tenant with Manager role can interact with all the environments
+	// (i.e. VMs) in a Workspace, as well as add new Tenants to the Workspace.
 	Manager WorkspaceUserRole = "manager"
-	// User allows to interact with owned vms
+	// User -> a Tenant with User role can only interact with his/her own
+	// environments (e.g. VMs) within that Workspace.
 	User WorkspaceUserRole = "user"
 )
 
-// UserWorkspaceData contains the info of the workspaces related to a user
-type UserWorkspaceData struct {
-	// reference to the workspace resource in the cluster
+// TenantWorkspaceEntry contains the information regarding one of the Workspaces
+// the Tenant is subscribed to, including his/her role.
+type TenantWorkspaceEntry struct {
+	// The reference to the Workspace resource the Tenant is subscribed to.
 	WorkspaceRef GenericRef `json:"workspaceRef"`
-	GroupNumber  uint       `json:"groupNumber,omitempty"`
-	// role of the user in the context of the workspace
+
+	// The role of the Tenant in the context of the Workspace.
 	Role WorkspaceUserRole `json:"role"`
+
+	// The number of the group the Tenant belongs to. Empty means no group.
+	GroupNumber uint `json:"groupNumber,omitempty"`
 }
 
-// TenantSpec defines the desired state of Tenant
+// TenantSpec is the specification of the desired state of the Tenant.
 type TenantSpec struct {
-	// Important: Run "make" to regenerate code after modifying this file
-
+	// The first name of the Tenant.
 	FirstName string `json:"firstName"`
 
+	// The last name of the Tenant.
 	LastName string `json:"lastName"`
 
+	// The email associated with the Tenant, which will be used to log-in
+	// into the system.
 	// +kubebuilder:validation:Pattern="^[a-zA-Z0-9.!#$%&'*+\\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$"
 	Email string `json:"email"`
 
-	// list of workspace the tenant subscribed to
-	Workspaces []UserWorkspaceData `json:"workspaces,omitempty"`
+	// The list of the Workspaces the Tenant is subscribed to, along with his/her
+	// role in each of them.
+	Workspaces []TenantWorkspaceEntry `json:"workspaces,omitempty"`
 
-	// list of the public keys of the tenant for SSH access
+	// The list of the SSH public keys associated with the Tenant. These will be
+	// used to enable to access the remote environments through the SSH protocol.
 	PublicKeys []string `json:"publicKeys,omitempty"`
 
-	// should the resource create the sandbox namespace for k8s practice environment
+	// Whether a sandbox namespace should be created to allow the Tenant play
+	// with Kubernetes.
 	// +kubebuilder:default=false
 	CreateSandbox bool `json:"createSandbox,omitempty"`
```

Comment:
_Sorry for the long comment_

_I mark this line, but there are more lines with the following issue:_

Placing the doc comment attached to the `//kubebuilder` marker, adds the marker inside the docs (you can see it if you hover over the filed on vscode). IMHO it is better not to have the marker in the docs. 

**current**
![wrong](https://user-images.githubusercontent.com/32385294/107873819-a5849e80-6eb5-11eb-9648-a324c39e80df.png)

**fixed**
![right](https://user-images.githubusercontent.com/32385294/107873820-a61d3500-6eb5-11eb-86f4-82ed412c8467.png)

The latter is produced with the following code:
```go
// +kubebuilder:default=false

// Whether a sandbox namespace should be created to allow the Tenant play
// with Kubernetes.
CreateSandbox bool `json:"createSandbox,omitempty"`
```

This code doesn't cancel the effect of the marker (it can be checked by search the word default in the generated CRD, it is still there)

Reply:
Thanks @GabriFila, did not know about this issue. Should be fixed now